File: backend/src/main/java/com/example/Flicktionary/global/security/CustomUserDetailsService.java
package com.example.Flicktionary.global.security;

import com.example.Flicktionary.domain.user.entity.UserAccount;
import com.example.Flicktionary.domain.user.service.UserAccountService;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;


@Service
@RequiredArgsConstructor
public class CustomUserDetailsService implements UserDetailsService {

    private final UserAccountService userAccountService;

    @Override
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        UserAccount userAccount = userAccountService.getUserByUsername(username);
        return new CustomUserDetails(userAccount);
    }
}

File: backend/src/main/kotlin/com/example/Flicktionary/global/security/CustomUserDetailsService.kt
package com.example.Flicktionary.global.security

import com.example.Flicktionary.domain.user.service.UserAccountService
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.stereotype.Service

@Service
class CustomUserDetailsService(
    private val userAccountService: UserAccountService
): UserDetailsService {

    override fun loadUserByUsername(username: String): UserDetails {
        val userAccount = userAccountService.getUserByUsername(username)
        return CustomUserDetails(userAccount)
    }
}